granulate_utils/metrics/api_client.py

```diff
@@ -0,0 +1,98 @@
+#
+# Copyright (c) Granulate. All rights reserved.
+# Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
+#
+import gzip
+import json
+from io import BytesIO
+from typing import IO, Any, Dict, List, Optional, cast
+
+import requests
+
+DEFAULT_API_SERVER_ADDRESS = "https://api.granulate.io"
+DEFAULT_REQUEST_TIMEOUT = 5
+DEFAULT_UPLOAD_TIMEOUT = 120
+
+
+class APIError(Exception):
+    def __init__(self, message: str, full_data: dict = None):
+        self.message = message
+        self.full_data = full_data
+
+    def __str__(self) -> str:
+        return self.message
+
+
+class APIClient:
```

Comment:
This replaces gProfiler's `BaseAPIClient`  or currently just a copy?

Reply:
I see you built it perfectly and gProfiler will be able to extend it with `log_bad_json` and `log_response`. Good job